lib/collection/src/collection/mod.rs

```diff
@@ -261,6 +269,10 @@ impl Collection {
 
         let locked_shard_holder = Arc::new(LockedShardHolder::new(shard_holder));
 
+        let local_stats_cache = LocalDataStatsCache::new_with_values(
+            Self::load_segment_stats(&locked_shard_holder).await,
+        );
```

Comment:
I agree
We could make this load optional for collections that have strict mode disabled. But since we can afford this check in the API, the performance overhead at collection initialization should not be too high.